src/parcsr_ls/par_mgr.c

```diff
@@ -6590,62 +6598,151 @@ hypre_MGRPrintCoarseSystem( void *mgr_vdata, HYPRE_Int print_flag)
+
+      if (info_path_length > 0)
+      {
+         if (myid)
+         {
+            info_path = hypre_TAlloc(char, info_path_length, HYPRE_MEMORY_HOST);
+         }
+      }
+      else
+      {
+         hypre_error_w_msg(HYPRE_ERROR_GENERIC, "Unable to create info path!");
+         return hypre_error_flag;
+      }
+      hypre_MPI_Bcast(info_path, info_path_length, hypre_MPI_CHAR, 0, comm);
+
+      /* Save info_path */
+      (mgr_data -> info_path) = info_path;
+   }
+   else
+   {
+      if (info_path)
+      {
+         info_path_length = strlen(info_path);
       }
-      hypre_printf("\n");
    }
-   hypre_printf("Number of Reserved Cpoints: %d\n", (mgr_data -> reserved_coarse_size));
-   hypre_printf("Keep reserved Cpoints to level: %d\n", (mgr_data -> lvl_to_keep_cpoints));
 
-   hypre_printf("\n MGR Solver Parameters: \n");
-   hypre_printf("Number of interpolation sweeps: %d\n", (mgr_data -> num_interp_sweeps));
-   hypre_printf("Number of restriction sweeps: %d\n", (mgr_data -> num_restrict_sweeps));
-   if (mgr_data -> level_smooth_type != NULL)
+   /* Print MGR parameters to file */
+   if (print_level & HYPRE_MGR_PRINT_INFO_PARAMS)
    {
-      hypre_printf("Global smoother type: %d\n", (mgr_data -> level_smooth_type)[0]);
-      hypre_printf("Number of global smoother sweeps: %d\n", (mgr_data -> level_smooth_iters)[0]);
+      /* TODO (VPM): print internal MGR parameters to file */
+
+      /* Signal that the MGR parameters have already been printed */
+      (mgr_data -> print_level) &= ~HYPRE_MGR_PRINT_INFO_PARAMS;
+      (mgr_data -> print_level) |= HYPRE_MGR_PRINT_RESERVED_A;
    }
-   hypre_printf("Max number of iterations: %d\n", (mgr_data -> max_iter));
-   hypre_printf("Stopping tolerance: %e\n", (mgr_data -> tol));
-   hypre_printf("Use default coarse grid solver: %d\n", (mgr_data -> use_default_cgrid_solver));
-   /*
-      if ((mgr_data -> fsolver_mode) >= 0)
+
+   /* Print linear system matrix and dofmap */
+   if ((print_level & HYPRE_MGR_PRINT_INFO_MATRIX) && par_A)
+   {
+      /* Build dofmap array */
+      dofmap = hypre_IntArrayCreate(hypre_ParCSRMatrixNumRows(par_A));
+      hypre_IntArrayInitialize_v2(dofmap, HYPRE_MEMORY_HOST);
+      if (point_marker_array)
       {
-         hypre_printf("Use AMG solver for full AMG F-relaxation: %d\n", (mgr_data -> fsolver_mode));
+         hypre_TMemcpy(hypre_IntArrayData(dofmap), point_marker_array,
+                       HYPRE_Int, hypre_ParCSRMatrixNumRows(par_A),
+                       HYPRE_MEMORY_HOST, HYPRE_MEMORY_HOST);
       }
-   */
+      else
+      {
+         hypre_IntArraySetInterleavedValues(dofmap, block_size);
+      }
+
+      /* Print Matrix */
+      hypre_ParPrintf(comm, "Writing matrix to path: %s\n", info_path);
+      filename = hypre_TAlloc(char, strlen(info_path) + 16, HYPRE_MEMORY_HOST);
+      hypre_sprintf(filename, "%s/IJ.out.A", info_path);
+      hypre_ParCSRMatrixPrintBinaryIJ(par_A, 0, 0, filename);
```

Comment:
When printing to file, are you only supporting binary files?  Based on my past experience interacting with users via binary files, it can sometimes be problematic and is the reason we have used ASCII.  Can we support both, but have binary be the default?

Reply:
Yes, that's a good idea. What about:
1. Rename `HYPRE_MGR_PRINT_INFO_MATRIX` to `HYPRE_MGR_PRINT_INFO_MATRIX_BIN`
2. Add `HYPRE_MGR_PRINT_INFO_MATRIX_ASCII`

Same for RHS

Reply:
(This way we don't need a separate flag to signal binary or ascii and we can handle all cases with `print_level` alone)

Reply:
Well, I was thinking of only using print_level anyway.  Wouldn't it be easier to just add a '32' to the print level to signify printing ASCII?  Otherwise, don't you need both a '32' and a '64'?

Reply:
Oh yes, now I see it. Agreed, will do this way

Reply:
Rob, I changed code "8" to signal ASCII printing and moved the matrix and RHS printout codes to "16" and "32". I also reserved "64" and "128" to print additional debugging information (to be implemented in another PR along with code "4"). See changes in [4562249](https://github.com/hypre-space/hypre/pull/976/commits/45622494720a6d691518559c3db9173d352f8d1b)